crates/nu-command/src/strings/encode_decode/hex.rs

```diff
@@ -0,0 +1,179 @@
+use crate::input_handler::{operate as general_operate, CmdArgument};
+use nu_engine::CallExt;
+use nu_protocol::ast::{Call, CellPath};
+use nu_protocol::engine::{EngineState, Stack};
+use nu_protocol::{PipelineData, ShellError, Span, Value};
+
+fn hex_decode(value: String) -> Result<Vec<u8>, ()> {
+    let mut res = Vec::with_capacity(value.len() / 2);
+    let mut chars = value.chars();
+    while let Some(c) = chars.next() {
+        let c = c.to_digit(16).ok_or(())?;
+        let c2 = chars.next().and_then(|c| c.to_digit(16)).ok_or(())?;
+        res.push((c * 16 + c2).try_into().map_err(|_| ())?);
+    }
+    Ok(res)
+}
```

Comment:
**Nitpicking alert**

From an API perspective this doesn't have to take the `String` and could work with `&str`

Would be curious how this compares to [`u8::from_str_radix`](https://doc.rust-lang.org/core/primitive.u8.html#method.from_str_radix).

See the version in the parser:
https://github.com/nushell/nushell/blob/79d07358643193d411b85b279644d3d1157a1161/crates/nu-parser/src/parser.rs#L1397-L1406